Providers/Profile.swift

```diff
@@ -798,8 +836,49 @@ public class BrowserProfile: Profile {
             }
         }
 
+        @objc func onStartSyncing(notification: NSNotification) {
+            syncDisplayState = .InProgress
```

Comment:
I agree about the lock. This was taken out accidentally somewhere along the lines.

As for `beginSyncing`, `onStartSyncing` was created and this code was moved here in response to another review comment some time ago from @sleroux who felt `beginSyncing` was the wrong place for it. I'm going to leave it like this.